### ISSUE TITLE
If you have GuardDoesNotMatch exception with inherited user classes...

### ISSUE THREAD

**Issue by scs75** [spatie/laravel-permission#358]
The context: My User class is a parent and has child classes like TenantUser, LandlordUser. 

The config/auth.php:
`'providers' => [
        'users' => [
            'driver' => 'multi_eloquent',
            'model' => App\Models\User::class,
        ],
 ],`

When I try 
`$tenant = new TenantUser(1);
$tenant->assignRole('manager');`
it throws a GuardDoesNotMatch exception.

Can fix this with a guard_name property on User class:
`protected $guard_name = 'admin';`

Just sharing if you have similar problem.

**Comment by ctf0** [comment]
u literally saved me from insanity.

**Comment by colinjo** [comment]
Didnt help, but found out something else. I'd moved my User model to another folder.
Make sure you have a proper model filled in your auth config file: auth.providers.users.model. If the class cant be found the Guard::getNames() function will result in an empty collection which throws an exception later on.

**Comment by renepardon** [comment]
I use a custom guard and auth provider and had to set `protected $guard_name = 'mycustomguard';` on the User entity to make it work because I received this error:

```
The given role or permission should use guard `` instead of `mycustomguard`.
```

**Comment by captainfodder** [comment]
I just encountered this exact same problem myself today, and after enough searching around I stumbled across this issue and it resolved it for me. There is another open issue of another user having the same problem here #1950. Both of us had the same situation where we were both trying to use the ldaprecord package for user authentication, but were both using the default User model that ships with Laravel. I'm not sure about everyone else here but I wonder if they also had a similar situation and were trying to use [ldaprecord](https://ldaprecord.com/docs/laravel/v2) as well. 

These are my packages:
laravel/framework => 9.8.1
directorytree/ldaprecord-laravel => 2.5.5
spatie/laravel-permission => 5.5.2

It seems like there is a missing piece of the documentation for sure, but I'm not exactly sure where it belongs or when this solution is appropriate to implement.
